In [1]:
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf
import numpy as np
import scipy.stats as ss

<h2>1) Using the posted data set WAGE2.DTA, estimate a model where you regress years of education (educ)
on the number of siblings (sibs), mother’s education (meduc), and father’s education (feduc).</h2>

<h3>(a) Write down the estimated regression equation.</h3>

𝑒𝑑𝑢𝑐̂ = 𝛽<sub>0</sub> + 𝛽<sub>1</sub> 𝑠𝑖𝑏𝑠 + 𝛽<sub>2</sub> 𝑚𝑒𝑑𝑢𝑐 + 𝛽<sub>3</sub> 𝑓𝑒𝑑𝑢𝑐
<br>
&emsp;    = 10.36426 − 0.0936359 𝑠𝑖𝑏𝑠 + 0.1307872 𝑚𝑒𝑑𝑢𝑐 + 0.2100041 𝑓𝑒𝑑𝑢𝑐

In [4]:
wage_df = pd.read_stata('WAGE2.dta')
wage_df = wage_df.dropna(subset=['meduc', 'feduc'])
wage_df.reset_index(drop=True, inplace=True)
wage_df.head(5)

wage  hours   IQ  KWW  educ  exper  tenure  age  married  black  south  \
0   769     40   93   35    12     11       2   31        1      0      0   
1   808     50  119   41    18     11      16   37        1      0      0   
2   825     40  108   46    14     11       9   33        1      0      0   
3   650     40   96   32    12     13       7   32        1      0      0   
4   562     40   74   27    11     14       5   34        1      0      0   

   urban  sibs  brthord  meduc  feduc     lwage  
0      1     1      2.0    8.0    8.0  6.645091  
1      1     1      NaN   14.0   14.0  6.694562  
2      1     1      2.0   14.0   14.0  6.715384  
3      1     4      3.0   12.0   12.0  6.476973  
4      1    10      6.0    6.0   11.0  6.331502

In [5]:
x = wage_df[['sibs', 'meduc', 'feduc']]
y = wage_df[['educ']]
x = sm.add_constant(x)

In [6]:
reg = sm.OLS(y,x).fit()
print(reg.summary())

                            OLS Regression Results                            
Dep. Variable:                   educ   R-squared:                       0.214
Model:                            OLS   Adj. R-squared:                  0.211
Method:                 Least Squares   F-statistic:                     65.20
Date:                Wed, 08 Sep 2021   Prob (F-statistic):           2.73e-37
Time:                        20:17:32   Log-Likelihood:                -1518.2
No. Observations:                 722   AIC:                             3044.
Df Residuals:                     718   BIC:                             3063.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         10.3643      0.359     28.910      0.0

<h3>(b) Is the sign on the parameter estimate for sibs what you would expect? Explain.</h3>

<p>Yes, I expected the sign on sibs to be negative and it turned out to be negative. One would expect the relationship between educ and sibs to be a negative correlation since it could be thought that it each family has limited income. The more children a family has, the less each child would get financial support needed to succeed academically from their parents since it must be divided among the children.</p>

<h3>(c) Holding meduc and feduc fixed, by how much does sibs have to increase to reduce predicted years
of education by one year (it is ok if you get a noninteger answer)?</h3>

<p>𝑒𝑑𝑢𝑐̂= 𝛽<sub>0</sub>+ 𝛽<sub>1</sub> 𝑠𝑖𝑏𝑠+ 𝛽<sub>2</sub> 𝑚𝑒𝑑𝑢𝑐+ 𝛽<sub>3</sub> 𝑓𝑒𝑑𝑢𝑐 </p>
&emsp; -1 = −0.0936 Δ𝑠𝑖𝑏𝑠
<br>
&emsp; 1/0.0936 = Δ𝑠𝑖𝑏𝑠

In [9]:
print(-1/-0.0936)

10.683760683760683


<h3>(d) Explain the interpretation of the coefficient on meduc and feduc.</h3>

<p>The coefficient of meduc is 0.13078, this implies that the effect of one more year of mother’s education would increase the child’s year of education by 0.13078 year, or about 13.08%. As for the coefficient of feduc, the coefficient is 0.2100. This implies that the effect of one more year of father’s education would increase years of a child’s education by 0.2100 year, or about 21%.</p>

<h3>(e) Suppose that individual A has no siblings and his mother and father each have 12 years of
education. Now suppose that individual B has no siblings and his mother and father each of 16
years of education. What is the predicted difference in years of education between individuals A
and B? Show your work.</h3>

In [26]:
ind_a = str(round(10.3643 + 0 + (0.1308)*(12) + (0.2100)*(12), 2))
ind_b = str(round(10.3643 + 0 + (0.1308)*(16) + (0.2100)*(16), 2))
a_b_diff = str(round(float(ind_a) - float(ind_b), 2))
print(f'Individual A: educ_hat = 10.3643 + 0 + (0.1308)(12) + (0.2100)(12) ≈ {ind_a}')
print(f'Individual B: educ_hat = 10.3643 + 0 + (0.1308)(16) + (0.2100)(16) ≈ {ind_b}')
print(f'Individual A - Individual B = {a_b_diff}')

Individual A: educ_hat = 10.3643 + 0 + (0.1308)(12) + (0.2100)(12) ≈ 14.45
Individual B: educ_hat = 10.3643 + 0 + (0.1308)(16) + (0.2100)(16) ≈ 15.82
Individual A - Individual B = -1.37


<p>The predicted difference in years of education between individual A and B is approximately 1.37 years, in which individual B is expected to have more education than individual A due to their parents have more education than parents of individual A.</p>

***

<h2>2) Use the posted data set HPRICE1.DTA to estimate the model
<br>
<br>
&emsp;    price = 𝛽<sub>0</sub> + 𝛽<sub>1</sub>sqrft + 𝛽<sub>2</sub>bdrms + u
<br>
<br>
where price is the house price measured in thousands of dollars.</h2>

In [28]:
hprice_df = pd.read_stata('HPRICE1.dta')
hprice_df.dropna(subset=['sqrft', 'bdrms'], inplace=True)
hprice_df.reset_index(drop=True, inplace=True)
hprice_df.head(5)

price      assess  bdrms  lotsize  sqrft  colonial    lprice   lassess  \
0  300.0  349.100006      4   6126.0   2438         1  5.703783  5.855359   
1  370.0  351.500000      3   9903.0   2076         1  5.913503  5.862210   
2  191.0  217.699997      3   5200.0   1374         0  5.252274  5.383118   
3  195.0  231.800003      3   4600.0   1448         1  5.273000  5.445875   
4  373.0  319.100006      4   6095.0   2514         1  5.921578  5.765504   

   llotsize    lsqrft  
0  8.720297  7.798934  
1  9.200593  7.638198  
2  8.556414  7.225482  
3  8.433811  7.277938  
4  8.715224  7.829630

In [29]:
x = hprice_df[['sqrft', 'bdrms']]
y = hprice_df[['price']]
x = sm.add_constant(x)

In [30]:
reg = sm.OLS(y,x).fit()
print(reg.summary())

                            OLS Regression Results                            
Dep. Variable:                  price   R-squared:                       0.632
Model:                            OLS   Adj. R-squared:                  0.623
Method:                 Least Squares   F-statistic:                     72.96
Date:                Wed, 08 Sep 2021   Prob (F-statistic):           3.57e-19
Time:                        21:02:30   Log-Likelihood:                -488.00
No. Observations:                  88   AIC:                             982.0
Df Residuals:                      85   BIC:                             989.4
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        -19.3150     31.047     -0.622      0.5

<h3>(a) Write down the estimated regression equation.</h3>

<p>𝑝𝑟𝑖𝑐𝑒̂ = 𝛽<sub>0</sub> + 𝛽<sub>1</sub> 𝑠𝑞𝑟𝑓𝑡 + 𝛽<sub>2</sub> 𝑏𝑑𝑟𝑚𝑠 
<br>
&emsp; = −19.315 + 0.1284362 𝑠𝑞𝑟𝑓𝑡 + 15.19819 bdrms</p>

<h3>(b) What is the estimated increase in price for a house with one or more bedrooms, holding square
footage constant?</h3>

In [31]:
coef_df = pd.read_html(reg.summary().tables[1].as_html(),header=0,index_col=0)[0]
sqrft_coef = coef_df['coef']['sqrft']
bdrms_coef = coef_df['coef']['bdrms']
const_coef = coef_df['coef']['const']

In [32]:
print('change in price depending on bedrooms if holding sqrft fixed is ')
print(bdrms_coef * 1000)

change in price depending on bedrooms if holding sqrft fixed is 
15198.2


<h3>(c) What is the estimated increase in price for a house with an additional bedroom that is 140 square
feet in size? Compare this to your answer in part (b).</h3>

In [33]:
print('estimated increase in price for a house with an additional bedroom that is 140 squarefeet in size is')
print(bdrms_coef + (sqrft_coef*140))

estimated increase in price for a house with an additional bedroom that is 140 squarefeet in size is
33.1742


<h3>(d) What percentage in the variation in price is explained by square footage and the number of
bedrooms?</h3>

<p>R<sup>2</sup> implies how much sqrft and bdrms explain the variation in price. The value of R<sup>2</sup> in this case is 0.632. By multiplying R2 by 100, we will get the percentage term, which is 63.20%</p>

<h3>(e) Suppose a house has sqrft = 2,438 and bdrms = 4: Find the predicted selling price for this house.</h3>

In [34]:
print('Suppose a house has sqrft= 2,438 and bdrms= 4, the predicted selling price of the house is')
print((const_coef + ((bdrms_coef * 4)+(sqrft_coef * 2438)))*1000)

Suppose a house has sqrft= 2,438 and bdrms= 4, the predicted selling price of the house is
354517.0


<h3>(f) The actual selling price for the house in part (e) was $300,000 (so price = 300). Find the residual
for this house. Does this suggest that the purchaser underpaid or overpaid for this particular
house?</h3>

In [35]:
print('The residual for this house is ')
print(300000-((const_coef + ((bdrms_coef * 4)+(sqrft_coef * 2438)))*1000))

The residual for this house is 
-54517.0


***

<h2>3) Use the posted data set RDCHEM.DTA to estimate the model
<br>
<br>
&emsp;   rdntens = 𝛽<sub>0</sub> + 𝛽<sub>1</sub>lsales + 𝛽<sub>2</sub>profmarg + u
<br>
<br>
where rdintens is R&D spending as a percent of total sales, lsales is the log of sales, and profmarg
are profits as a percent of sales.</h2>

In [36]:
rdchem_df = pd.read_stata('RDCHEM.dta')
rdchem_df.head(5)

rd        sales     profits  rdintens   profmarg       salessq  \
0  430.600006  4570.200195  186.899994  9.421906   4.089536  2.088673e+07   
1   59.000000  2830.000000  467.000000  2.084806  16.501766  8.008900e+06   
2   23.500000   596.799988  107.400002  3.937668  17.995979  3.561702e+05   
3    3.500000   133.600006   -4.300000  2.619760  -3.218563  1.784896e+04   
4    1.700000    42.000000    8.000000  4.047619  19.047619  1.764000e+03   

     lsales       lrd  
0  8.427312  6.065180  
1  7.948032  4.077538  
2  6.391582  3.157000  
3  4.894850  1.252763  
4  3.737670  0.530628

In [154]:
rdchem_df.dropna(subset=['lsales', 'profmarg'], inplace=True)
rdchem_df.reset_index(drop=True, inplace=True)
x = rdchem_df[['lsales', 'profmarg']]
y = rdchem_df[['rdintens']]
x = sm.add_constant(x)
reg = sm.OLS(y,x).fit()
print(reg.summary())

                            OLS Regression Results                            
Dep. Variable:               rdintens   R-squared:                       0.098
Model:                            OLS   Adj. R-squared:                  0.036
Method:                 Least Squares   F-statistic:                     1.584
Date:                Sun, 01 Aug 2021   Prob (F-statistic):              0.222
Time:                        16:40:52   Log-Likelihood:                -63.333
No. Observations:                  32   AIC:                             132.7
Df Residuals:                      29   BIC:                             137.1
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.4723      1.676      0.282      0.7

<h3>(a) Test the hypothesis that R&D intensity (rdintens) does not change with sales against the alternative that it does increase with sales. Do the tests at the 5% and 10% levels.</h3>

In [148]:
t_crit = ss.t.ppf(df=29, q=0.05)
print('t critical value at 5% is ')
print(abs(t_crit))
t_crit = ss.t.ppf(df=29, q=0.10)
print('t critical value at 10% is ')
print(abs(t_crit))

t critical value at 5% is 
1.6991270265334977
t critical value at 10% is 
1.311433650507697


<p>The t statistics for lsales is 1.491, while the t critical value for 5% is approximately 1.699, and the t critical value for 10% is approximately 1.311. Since |tc| > |tb| at 5% level, we fail to reject the null hypothesis at 5%, but we reject the null hypothesis at 10% </p>

<h3>(b) Interpret the coefficient on profmarg. Is it economically signi…cant?</h3>

<p>The coefficient for profmarg is only .0500367, which is approximately about 5%. So it is not very economically significant</p>

<h3>(c) Does profmarg have a statistically significant effect on rdintens?</h3>

<p>|tprofmarg| = 1.093, which is lower than |tc| for both 5% and 10% levels, so it is not statistically significant</p>